### PR TITLE
reverse proxy setup with production webpack

### PR DESCRIPTION
**Pull request by manedurphy**
@jhou12 @pauly-van @slava000 

Here is my reverse proxy setup for the first service. I've also setup a production webpack config file set to bundle all services.

### CODE REVIEW

File: webpack.common.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const { resolve } = require('path');
+
```
Comment (jhou12) [inline webpack.common.js:4]: @manedurphy this is cool! So is the css and html plugin the only difference between the standard webpack we've been using and the prod? Do you think the pages would render the same way if someone else was just using regular webpack or should we all use the prod version?

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>FEC Corgis</title>
+</head>
+
+<body>
+    <div id="root"></div>
+    <div id="root2"></div>
```
Comment (jhou12) [inline public/index.html:12]: @manedurphy Are these divs for your primary and secondary module? How are you connecting them to your actual repos modules? I was looking for another reference to root1 and root2 and couldn't find it in the server routes. Did you just rename the getElementById reference to 'root' in your index.jsx?

**Review by jhou12 — COMMENTED** [review]
@manedurphy Really clean and simple setup, looking forward to hearing about the prod webpack you designed!

File: webpack.common.js (hunk 1 above)
Comment (manedurphy) [reply webpack.common.js:4]: I think we are all going to end up having a production configuration where the first step is to set the mode property to "production". This way we will see warnings when our bundle sizes exceed a certain threshold. I currently have it set such that the HtmlWebpackPlugin targets a template html file in my Proxy repo (in the public folder). That's just the template that webpack references when I run the build script. I have two root divs in their with unique IDs, where my first service will be renders in "root", and my second service will be rendered in "root2". When I clone everyone else's repo, I will add more root divs  and adjust the target ID that react-dom inserts each service in to.

**Pull request merged**